src/lib/y2users/linux/writer.rb

```diff
@@ -158,6 +165,66 @@ def add_user(user, issues)
         log.error("Error creating user '#{user.name}' - #{e.message}")
       end
 
+      # Command for creating new groups
+      GROUPADD = "/usr/sbin/groupadd".freeze
+      private_constant :GROUPADD
+
+      # Executes the command for creating the group
+      #
+      # @param group [Y2User::Group] the group to be created on the system
+      # @param issues [Y2Issues::List] a collection for adding an issue if something goes wrong
+      def add_group(group, issues)
+        args = [GROUPADD]
+        args << "--gid" << group.gid if group.gid
+        # TODO: system groups?
+        Yast::Execute.on_target!(args)
+      rescue Cheetah::ExecutionFailed => e
+        issues << Y2Issues::Issue.new(
+          format(_("The group '%{groupname}' could not be created"), groupname: group.name)
+        )
+        log.error("Error creating group '#{group.name}' - #{e.message}")
+      end
+
+      USERMOD_ATTRS = [:home, :shell, :gecos].freeze
+
+      def modify(new_user, old_user, issues)
+        change_password(new_user, issues) if new_user.password != old_user.password
+
+        usermod_changes = USERMOD_ATTRS.any? do |attr|
+          !new_user.public_send(attr).nil? &&
```

Comment:
NP: Remember that by now `gecos` is never going to be `nil`.

Reply:
yep, we should probably change that as I write above.